servers/graphql-kotlin-server/src/main/kotlin/com/expediagroup/graphql/server/extensions/collectionExtensions.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2022 Expedia, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.expediagroup.graphql.server.extensions
+
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.supervisorScope
+
+suspend fun <A, B> Iterable<A>.concurrentMap(
```

Comment:
while this is pretty generic, I wonder whether this should be exposed as part of the lib (currently it is public) -> maybe it should be internal (or private method in `GraphQLServer`)?

Reply:
Iets move it inside the `GraphQLServer` - this should not be a public API